### PR TITLE
add version upgrade

### PR DESCRIPTION
**Pull request by AllyW**

### CODE REVIEW

File: azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        version_arr = [self.major, ".", self.minor, ".", self.patch]
+        if self.pre:
+            version_arr.append(self.pre)
+            version_arr.append(self.pre_num)
+        return "".join([str(item) for item in version_arr])
+
+
+class VersionUpgradeMod:
+
+    def __init__(self, module_name, current_version, is_preview, is_experimental,
```
Comment (kairu-ms) [inline azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py]: Is there any reason to keep `is_experimental`?

File: azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                 meta_diff_before, meta_diff_after, next_version_pre_tag=None, next_version_segment_tag=None):
+        self.module_name = module_name
+        self.version = parse(current_version)
```
Comment (kairu-ms) [inline azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py]: It's better to verify all the existing extension versions can be parsed without error.

File: azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py (hunk 1 above)
Comment (AllyW) [reply azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py]: This is passed from current raw metadata and used to check whether to insert isPreview tag to next extension version for old cli users.

File: azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py (hunk 2 above)
Comment (AllyW) [reply azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py]: Yes, if version cannot be parsed, extension module installation will raise error too.

File: azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+    def __str__(self):
+        version_arr = [self.major, ".", self.minor, ".", self.patch]
+        if self.pre:
+            version_arr.append(self.pre)
+            version_arr.append(self.pre_num)
+        return "".join([str(item) for item in version_arr])
+
+
+class VersionUpgradeMod:
```
Comment (kairu-ms) [inline azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py]: I think version upgrade code should be in `azdev` instead of `azure-cli-diff-tool`.

Comment (AllyW) [reply azure-cli-diff-tool/azure_cli_diff_tool/version_upgrade.py]: Version upgrade is working based on diff tool, and in extension ci, version calculation is planned to be used as func call, so packaged with meta-diff together. azdev can expose related cmd later if needed.

File: HISTORY.rst [hunk 4]
```diff
@@ -2,6 +2,9 @@
 
 Release History
 ===============
+0.1.62
+++++++
+* `azdev extension cal-next-version`: Add some examples.
```
Comment (kairu-ms) [inline HISTORY.rst]:
```suggestion
* `azdev extension cal-next-version`: New command to calculate valid version for next extension module release..
```

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**